README
WWW-Google-Places

The README  is  used  to  introduce the module and provide instructions on how to
install the module, any machine dependencies it may have (for example C compilers
and installed libraries) and any other information that should be provided before
the module is installed.

A  README  file  is required for CPAN modules since CPAN extracts the README file
from a module distribution so that people browsing the archive can use it to  get
an  idea  of  the  module's  uses.  It  is usually a good idea to provide version
information here so that people can decide whether fixes for the module are worth
downloading.

INSTALLATION

To install this module, run the following commands:

	perl Makefile.PL
	make
	make test
	make install

SUPPORT AND DOCUMENTATION

After installing,  you  can  find  documentation for this module with the perldoc
command.

    perldoc WWW::Google::Places

You can also look for information at:

    RT, CPAN's request tracker (report bugs here)
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=WWW-Google-Places

    AnnoCPAN, Annotated CPAN documentation
        http://annocpan.org/dist/WWW-Google-Places

    CPAN Ratings
        http://cpanratings.perl.org/d/WWW-Google-Places

    Search CPAN
        http://search.cpan.org/dist/WWW-Google-Places/

LICENSE AND COPYRIGHT

Copyright (C) 2011 - 2015 Mohammad S Anwar

This program is free software; you can redistribute it and/or modify it under the
terms of the the Artistic License (2.0).You may obtain a copy of the full license
at:

L<http://www.perlfoundation.org/artistic_license_2_0>

Any  use,  modification, and distribution of the Standard or Modified Versions is
governed by this Artistic License.By using, modifying / distributing the Package,
you accept this license. Do not use, modify, or distribute the Package, if you do
not accept this license.

If your Modified Version has been derived from a Modified Version made by someone
other than you, you're nevertheless required to ensure that your Modified Version
complies with the requirements of this license.

This  license  does  not  grant you the right to use any trademark, service mark,
tradename, or logo of the Copyright Holder.

This license includes the non-exclusive, worldwide, free-of-charge patent license
to  make,  have made, use, offer to sell, sell, import and otherwise transfer the
Package with respect to any patent claims licensable by the Copyright Holder that
are  necessarily  infringed  by the  Package.  If you institute patent litigation
(including  a  cross-claim or  counterclaim)  against any party alleging that the
Package constitutes direct or contributory patent infringement,then this Artistic
License to you shall terminate on the date that such litigation is filed.

Disclaimer  of  Warranty:  THE  PACKAGE  IS  PROVIDED BY THE COPYRIGHT HOLDER AND
CONTRIBUTORS  "AS IS'  AND WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES. THE IMPLIED
WARRANTIES    OF     MERCHANTABILITY,   FITNESS   FOR   A PARTICULAR  PURPOSE, OR
NON-INFRINGEMENT ARE DISCLAIMED TO THE EXTENT PERMITTED BY YOUR LOCAL LAW. UNLESS
REQUIRED BY LAW, NO COPYRIGHT HOLDER OR CONTRIBUTOR WILL BE LIABLE FOR ANY DIRECT,
INDIRECT, INCIDENTAL, OR CONSEQUENTIAL DAMAGES ARISING IN ANY WAY OUT OF THE USE
OF THE PACKAGE,EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.


Changes
Revision history for WWW::Google::Places

0.31   Sun Jul 31 03:30:00 2016
       - Added package for attributes 'photos' and 'address_components'.
       - Updated method WWW::Google::Place::Review::as_string().
       - Added method WWW::Google::Place::DetailResult::as_string().
       - Added method WWW::Google::Place::SearchResult::as_string().
       - Updated section "SYNOPSIS" to reflect the above changes.
       - Tidied up unit test scripts.

0.30   Sun Jul 10 12:00:00 2016
       - Updated copyright year information.
       - Addressed issue raised in the CPAN Tester Report.
         http://www.cpantesters.org/cpan/report/3ed22692-434c-11e6-b35e-6fe162f69d7e

0.29   Sun Dec 13 03:15:00 2015
       - Fixed issue RT: 110292 raised by SREZIC.

0.28   Sat Dec 12 05:05:00 2015
       - Used new package Method::ParamValidator for validating method parameters dynamically.

0.27   Mon Oct 05 10:50:00 2015
       - Upgraded to use WWW::Google::UserAgent v0.20.

0.26   Wed Sep 30 10:10:00 2015
       - Upgraded to use WWW::Google::UserAgent v0.19.

0.25   Sun Sep 06 08:35:00 2015
       - Added link to the official Google API document.

0.24   Wed Aug 19 10:10:00 2015
       - Upgraded to use WWW::Google::UserAgent v0.18.

0.23   Sun Aug 09 12:00:00 2015
       - Upgraded to use WWW::Google::UserAgent v0.17.

0.22   Tue May 19 10:40:00 2015
       - Added missing 'pagetoken' field to $FIELDS in the package WWW::Google::Places::Params.

0.21   Sun May 17 08:30:00 2015
       - Fixed setting default values to class attributes.
       - Made the return value of method search() and paged_search() based on wantarray().
       - Upgraded to use WWW::Google::UserAgent v0.16.

0.20   Mon May 04 18:15:10 2015
       - Fixed sample code in the pod document.

0.19   Mon Apr 27 13:40:10 2015
       - Added following packages
         - WWW::Google::Places::Geometry
         - WWW::Google::Places::Location
         - WWW::Google::Places::Review
         - WWW::Google::Places::Review::Aspect
       - Updated package WWW::Google::Places::DetailResult to respect the above changes.
       - Updated package WWW::Google::Places::SearchResult to respect the above changes.
       - Added section "SYNOPSIS" to the pod document of WWW::Google::Places.

0.18   Fri Feb 13 17:20:10 2015
       - Upgraded to use Test::Internet v0.04.
       - Updated test script 05-paging.t

0.17   Thu Feb 12 23:00:10 2015
       - Used Test::Internet in the test script 05-paging.t
       - Tidied up pod document in general.

0.16   Wed Feb 11 11:40:10 2015
       - Merged pull request from Hunter McMillen (mcmillhj) about new method paged_search().
       - Tidied up test script 05-paging.t.
       - Upgraded to use WWW::Google::UserAgent v0.14.
       - General code tidied up.

0.15   Tue Jan 13 20:00:10 2015
       - Updated description of methods in the pod document of the package WWW::Google::Places.

0.14   Mon Jan 12 15:05:18 2015
       - Upgraded to use WWW::Google::UserAgent v0.13.
       - Updated the description of the following methods in the pod document:
         - search()
         - details()
         - add()
         - delete()
       - Marked the following methods as DEPRECATED/UNSUPPORTED in the pod document:
         - search_place()
         - place_detail()
         - add_place()
         - delete_place()
         - place_checkins()

0.13    Mon Jan 12 12:40:18 2015
        - Added 'use 5.006;' line to the following test scripts:
          - t/01-new.t
          - t/02-search_place.t
          - t/03-place_detail.t
          - t/04-add_place.t
          - t/meta-json.t
          - t/meta-yml.t

0.12    Sat Jan 10 18:50:18 2015
        - Upgraded to use WWW::Google::UserAgent v0.12.
        - Updated Makefile.PL to generate the key 'provides' for META file correctly.
        - Documented methods in the package WWW::Google::Places::SearchResult and WWW::Google::Places::DetailResult.
        - Removed SIGNATURE file.
        - Added unit test to test META files.
        - Added LICENSE file.
        - Updated Copyright year information.

0.11    Sat Aug 30 01:35:18 2014
        - Added section "REPOSITORY" to the pod document.
        - Added key "META_MERGE" to Makefile.PL file.

0.10    Sun Aug 10 01:16:00 2014
        - Tidied up README file.
        - Updated Copyright year details.

0.09    Tue Jul 29 15:35:20 2014
        - Updated Makefile.PL to make the pre-requisite WWW::Google::UserAgent v0.07

0.08    Tue Jul 29 00:38:00 2014
        - Updated Makefile.PL to make the pre-requisite WWW::Google::UserAgent v0.06

0.07    Mon Jul 28 14:35:00 2014
        - Updated Makefile.PL to make the pre-requisite WWW::Google::UserAgent v0.03

0.06    Sun Jul 27 11:51:25 2014
        - Made use of WWW::Google::UserAgent v0.02.
        - Deleted Build.PL file.

0.05    Sun Jul 07 16:27:20 2014
        - Restructured to use the common library WWW::Google::UserAgent

0.04    Mon Aug 08 09:46:11 2011
        - Made Readonly v1.03 as minimum requirement.
        - Prepared the distribution using the Module::Signature v0.68.

0.03    Wed May 18 20:00:00 2011
        - Updated text for "LICENSE AND COPYRIGHT".

0.02    Mon May 16 13:09:00 2011
        - Updated the CONSTRUCTOR logic, so that it can now accept the api_key and sensor param
          as list i.e new($api_key, $sensor) as well.
        - Minor pod changes.

0.01    Mon May 16 11:30:01 2011
        - First version, released.


Makefile.PL
#!/usr/bin/perl

use 5.006;
use strict;
use warnings FATAL => 'all';
use ExtUtils::MakeMaker;
use File::ShareDir::Install;

install_share dist => 'share';

WriteMakefile(
    NAME               => 'WWW::Google::Places',
    AUTHOR             => q{Mohammad S Anwar <mohammad.anwar@yahoo.com>},
    VERSION_FROM       => 'lib/WWW/Google/Places.pm',
    ABSTRACT_FROM      => 'lib/WWW/Google/Places.pm',
    LICENSE            => 'artistic_2',
    MIN_PERL_VERSION   => 5.006,
    CONFIGURE_REQUIRES => {
        'ExtUtils::MakeMaker'     => 0,
        'File::ShareDir::Install' => 0,
    },
    BUILD_REQUIRES       => {
        'Test::More'     => '1.001014',
        'Test::Internet' => '0.04',
    },
    PREREQ_PM => {
        'Moo'                    => '2.000000',
        'namespace::clean'       => '0.24',
        'File::Share'            => '0.25',
        'JSON'                   => '2.53',
        'WWW::Google::UserAgent' => '0.20',
        'Method::ParamValidator' => '0.05',
    },
    dist  => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
    clean => { FILES => 'WWW-Google-Places-*' },
    (eval { ExtUtils::MakeMaker->VERSION(6.46) } ? (META_MERGE => {
        'meta-spec' => { version => 2 },
        provides    => {
            'WWW::Google::Places'                 => { file => 'lib/WWW/Google/Places.pm',               version => '0.31' },
            'WWW::Google::Places::Params'         => { file => 'lib/WWW/Google/Places/Params.pm',        version => '0.31' },
            'WWW::Google::Places::SearchResult'   => { file => 'lib/WWW/Google/Places/SearchResult.pm',  version => '0.31' },
            'WWW::Google::Places::DetailResult'   => { file => 'lib/WWW/Google/Places/DetailResult.pm',  version => '0.31' },
            'WWW::Google::Places::Geometry'       => { file => 'lib/WWW/Google/Places/Geometry.pm',      version => '0.31' },
            'WWW::Google::Places::Location'       => { file => 'lib/WWW/Google/Places/Location.pm',      version => '0.31' },
            'WWW::Google::Places::Photo'          => { file => 'lib/WWW/Google/Places/Photo.pm',         version => '0.31' },
            'WWW::Google::Places::Address'        => { file => 'lib/WWW/Google/Places/Address.pm',       version => '0.31' },
            'WWW::Google::Places::Review'         => { file => 'lib/WWW/Google/Places/Review.pm',        version => '0.31' },
            'WWW::Google::Places::Review::Aspect' => { file => 'lib/WWW/Google/Places/Review/Aspect.pm', version => '0.31' },
        },
        resources => {
            repository => {
                type => 'git',
                url  => 'https://github.com/manwar/WWW-Google-Places.git',
                web  => 'https://github.com/manwar/WWW-Google-Places',
            },
        }})
     : ()
    ),
);


package MY;
use File::ShareDir::Install 'postamble';
